### PR TITLE
DZ10

### PR DESCRIPTION
**Pull request by Olga8878**

### CODE REVIEW

File: src/Lesson10/Task1.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            if (shortestWord.equalsIgnoreCase("") || word.length() < shortestWord.length()) {
+                shortestWord = word;
+            }
+            if (longestWord.equalsIgnoreCase("") || word.length() >= longestWord.length()) {
+                longestWord = word;
+            }
+        }
+
+        String lastShortestWord = "";
+        for (String word : words) {
```
Comment (DHrytsuk) [inline src/Lesson10/Task1.java:20]: I think, this and the loop below are not necessary and we can do everuthing in one loop

File: src/Lesson10/Task1.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        System.out.println("Введите строку:");
+        String input = scanner.nextLine();
+
+        String[] words = input.split("\\s+");
+
+        String shortestWord = "";
+        String longestWord = "";
+
+        for (String word : words) {
+            if (shortestWord.equalsIgnoreCase("") || word.length() < shortestWord.length()) {
```
Comment (DHrytsuk) [inline src/Lesson10/Task1.java]: shortestWord.equalsIgnoreCase("") can be removed

File: src/Lesson10/Task3.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+        System.out.println("Введите строку:");
+        String input = scanner.nextLine();
+        System.out.println("Введи номер слова:");
+        int word = scanner.nextInt();
+
+        String[] words = input.split("\\s+");
+
+        if (word > 0 & word <= words.length) {
+            String word1 = words[word - 1];
```
Comment (DHrytsuk) [inline src/Lesson10/Task3.java]: Try to avoid such naming, like `word1`

File: src/Lesson10/Task4.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    public static void main(String[] args) {
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Введите строку:");
+        String input = scanner.nextLine();
+
+        StringBuilder duplicatedStr = new StringBuilder();
+
+        for (int i = 0; i < input.length(); i++) {
+            char Char1 = input.charAt(i);
```
Comment (DHrytsuk) [inline src/Lesson10/Task4.java]: Same about naming

File: src/Lesson10/Task1.java (hunk 1 above)
Comment (DHrytsuk) [reply src/Lesson10/Task1.java:20]: What about this comment?
Still not fixed

File: src/Lesson10/Task1.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        System.out.println("Введите строку:");
+        String input = scanner.nextLine();
+
+        String[] words = input.split("\\s+");
+
+        String shortestWord = "";
+        String longestWord = "";
+
+        for (String word : words) {
+            if (shortestWord.isEmpty() || word.length() < shortestWord.length()) {
```
Comment (DHrytsuk) [inline src/Lesson10/Task1.java]: Could you implement your logic without `shortestWord.isEmpty()`?

File: src/Lesson10/Task1.java (hunk 1 above)
Comment (Olga8878) [reply src/Lesson10/Task1.java:20]: Он исправлен.

Comment (DHrytsuk) [reply src/Lesson10/Task1.java:20]: But why do you still have this loop?

File: src/Lesson10/Task1.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Введите строку:");
+        String input = scanner.nextLine();
+
+        String[] words = input.split("\\s+");
+
+        String shortestWord = words[0];
+        String longestWord = words[0];
+
+        String lastShortestWord = "";
```
Comment (DHrytsuk) [inline src/Lesson10/Task1.java:17]: Why do you have both shortestWord and shortestWord?
Isn't it the same?

**Pull request merged**